notebook/serverextensions.py

```diff
@@ -0,0 +1,203 @@
+        for arg in self.extra_args:
+            if self.python:
+                self.toggle_server_extension_python(arg)
+            else:
+                self.toggle_server_extension(arg)
+
+
+class EnableServerExtensionApp(ToggleServerExtensionApp):
+
+    name = "jupyter serverextension enable"
+    description = "Enable a server extension using frontend configuration files."
+    _toggle_value = True
+
+
+class DisableServerExtensionApp(ToggleServerExtensionApp):
+
+    name = "jupyter serverextension disable"
+    description = "Disable an serverextension using frontend configuration files."
+    _toggle_value = False
+
+
+class ListServerExtensionsApp(BaseNBExtensionApp):
+
+    name = "jupyter serverextension list"
+    version = __version__
+    description = "List all server extensions known by the configuration system"
+
+    def list_server_extensions(self):
+        config_dirs = jupyter_config_path()
+        for config_dir in config_dirs:
+            self.log.info('config dir: {}'.format(config_dir))
+            cm = BaseJSONConfigManager(parent=self, config_dir=config_dir)
+            data = cm.get("jupyter_notebook_config")
+            server_extensions = (
+                data.setdefault("NotebookApp", {})
+                .setdefault("nbserver_extensions", {})
+            )
+            for x in server_extensions:
+                self.log.info('    {1} {0}'.format(x, GREEN_ENABLED if server_extensions[x] else RED_DISABLED))
+
+    def start(self):
+        self.list_server_extensions()
+
+
+_examples = """
+jupyter serverextension list                            # list all configured nbextensions
+jupyter serverextension enable --py <packagename>   # enable all nbextensions in a Python package
+jupyter serverextension disable --py <packagename>  # disable all nbextensions in a Python package
+"""
+
+
+class ServerExtensionApp(BaseNBExtensionApp):
+
+    name = "jupyter serverextension"
+    version = __version__
+    description = "Work with Jupyter server extensions"
+    examples = _examples
+
+    subcommands = dict(
+        enable=(EnableServerExtensionApp, "Enable an server extension"),
+        disable=(DisableServerExtensionApp, "Disable an server extension"),
+        list=(ListServerExtensionsApp, "List server extensions")
+    )
+
+    def start(self):
+        super(ServerExtensionApp, self).start()
+
+        # The above should have called a subcommand and raised NoStart; if we
+        # get here, it didn't, so we should self.log.info a message.
+        subcmds = ", ".join(sorted(self.subcommands))
+        sys.exit("Please supply at least one subcommand: %s" % subcmds)
+
+main = ServerExtensionApp.launch_instance
+
+# ------------------------------------------------------------------------------
+# Private API
+# ------------------------------------------------------------------------------
+
+
+def _get_server_extension_metadata(package):
+    m = __import__(package)
```

Comment:
I think I've missed that discussion too.  But yes, you can have multiple extensions in one package, but the definition resides in one namespace as a list of extension definitions.